### PR TITLE
Add support for custom coala notifications

### PR DESCRIPTION
**Pull request by frextrite**
Add support for custom coala notifications

### CODE REVIEW

File: src/main/java/io/coala/jetbrains/analysis/CodeAnalysisTask.java [hunk 1]
```diff
@@ -1,11 +1,15 @@
 package io.coala.jetbrains.analysis;
 
 import com.intellij.execution.ExecutionException;
+import com.intellij.execution.process.ProcessNotCreatedException;
 import com.intellij.execution.process.ProcessOutput;
+import com.intellij.notification.*;
```
Comment (ksdme) [inline src/main/java/io/coala/jetbrains/analysis/CodeAnalysisTask.java]: Everything else uses named imports, so this should as well be done that way.

File: src/main/java/io/coala/jetbrains/analysis/CodeAnalysisTask.java [hunk 2]
```diff
@@ -32,17 +36,17 @@ public void run(@NotNull ProgressIndicator progressIndicator) {
...
-        progressIndicator.setFraction(.8);
-        progressIndicator.setText("coala analysis done!");
+            progressIndicator.setIndeterminate(false);
+            progressIndicator.setFraction(.8);
+            progressIndicator.setText("coala analysis done!");
 
-        // TODO: Parse the JSON string, show errors/annotate
+            // TODO: Parse the JSON string, show errors/annotate
+        } catch (ExecutionException e) {
+            Notifier.showErrorNotification("Failed to run coala. Make sure the supplied path is correct and .coafile exists in project root.", e);
```
Comment (ksdme) [inline src/main/java/io/coala/jetbrains/analysis/CodeAnalysisTask.java]: This statement is `147` characters long.

File: src/main/java/io/coala/jetbrains/utils/Notifier.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
+package io.coala.jetbrains.utils;
+
+import com.intellij.notification.*;
```
Comment (ksdme) [inline src/main/java/io/coala/jetbrains/utils/Notifier.java]: Same as mentioned above (or below).

File: src/main/java/io/coala/jetbrains/utils/Notifier.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.ui.MessageType;
+import com.intellij.openapi.wm.ToolWindowManager;
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class Notifier {
+    private static final String title = "coala";
+    private static final NotificationGroup NOTIFICATION = new NotificationGroup("coala", NotificationDisplayType.BALLOON, true);
```
Comment (ksdme) [inline src/main/java/io/coala/jetbrains/utils/Notifier.java]: `129` Characters long, max is `120`.

File: src/main/java/io/coala/jetbrains/utils/Notifier.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        final Notification notification = NOTIFICATION.createNotification(title, message, NotificationType.INFORMATION, null);
+        showNotification(notification);
+    }
+
+    public static void showWarningNotification(String message) {
+        final Notification notification = NOTIFICATION.createNotification(title, message, NotificationType.WARNING, null);
+        showNotification(notification);
+    }
+
+    public static void showErrorNotification(String message, @Nullable Throwable throwable) {
```
Comment (ksdme) [inline src/main/java/io/coala/jetbrains/utils/Notifier.java]: Needs code refactoring, I cannot comprehend any of this.

File: src/main/java/io/coala/jetbrains/utils/Notifier.java (hunk 4 above)
Comment (frextrite) [reply src/main/java/io/coala/jetbrains/utils/Notifier.java]: Lines flowing over 120, to about 125-130 have been kept as it is to improve code readability.

**Pull request merged**